### ISSUE TITLE
Add "get_singleton(singleton_name)" to get a singleton by name.

### ISSUE THREAD

**Issue by KnightNine** [godotengine/godot-proposals#9984]
### Describe the project you are working on

A function which calls any function across all remote players' games so I don't have to add an rpc function for every function in my project.

### Describe the problem or limitation you are having in your project

I cannot get the singleton containing the function(s) in the remote player's game without adding jank that is susceptible to user error:

I have these functions (in my singleton to do with networking) to share a function call between players within various singletons:

```gdscript
#Function sharing===============================
var AUTOLOAD_SCRIPTS = {
	"System":System,
	"Global_Functions_3D":Global_Functions_3D,
	"Global_Menu_Functions":Global_Menu_Functions,
}

@rpc("any_peer","reliable","call_local")
func rpc_make_synchronised_call(singleton_name,function_name, params_arr):
	var caller = Callable(AUTOLOAD_SCRIPTS[singleton_name],function_name)
	caller.callv(params_arr)

func make_synchronised_call(singleton_name, function_name, params_arr):
	rpc_make_synchronised_call.rpc(singleton_name, function_name, params_arr)
```

But I have to manually update the `AUTOLOAD_SCRIPTS` if I add a new singleton for this function to work which is lame.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Adds a way to get the autoload scripts by name.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

basically this:
```
var caller = Callable(get_singleton(singleton_name),function_name)
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Yes but it's annoying knowing that I am copying data that should probably be exposed in the editor. 

### Is there a reason why this should be core and not an add-on in the asset library?

Is there an addon for this?

**Comment by AThousandShips** [comment]
Like [`Engine.get_singleton`](https://docs.godotengine.org/en/stable/classes/class_engine.html#class-engine-method-get-singleton)?

**Comment by KnightNine** [comment]
@AThousandShips 

yes exactly that.... but Engine.get_singleton() seems to not work at all in 4.2.2-stable.  Just always returns null.💀

**Comment by AThousandShips** [comment]
Then please create an bug report instead, but first make sure the singleton exists, by `has_singleton` and `get_singleton_list` to make sure you got the name right etc.

**Comment by KnightNine** [comment]
get_singleton_list() doesn't include The Autoload Scripts (which are also called singletons in the docs) defined in my project settings. 
```
["Performance", "TextServerManager", "PhysicsServer2DManager", "PhysicsServer3DManager", "NavigationMeshGenerator", "ProjectSettings", "IP", "Geometry2D", "Geometry3D", "ResourceLoader", "ResourceSaver", "OS", "Engine", "ClassDB", "Marshalls", "TranslationServer", "Input", "InputMap", "EngineDebugger", "Time", "GDExtensionManager", "ResourceUID", "WorkerThreadPool", "ThemeDB", "JavaClassWrapper", "JavaScriptBridge", "DisplayServer", "RenderingServer", "AudioServer", "PhysicsServer2D", "PhysicsServer3D", "NavigationServer2D", "NavigationServer3D", "XRServer", "CameraServer"]
```
anyways ill add a bug report.

**Comment by AThousandShips** [comment]
They probably won't be registered there so I think that's a documentation issue not a bug

**Comment by Mickeon** [comment]
I explicitly added that note to address this confusion.
